### DISCUSSION TITLE
Ability to hook into transactions

### DISCUSSION THREAD

**Discussion by georeith** [mobxjs/mobx#3106] [Ideas]
I am using Mobx alongside other libraries such as https://github.com/yjs/yjs, these tools also want to be aware of transactions / batching.

I could expose a single function that people can use which combines transactions from both tools however there are so many entry points to transactions in mobx (`action`, `runInAction`, `action.bound`...) it would be simpler if I could just hook into the base transaction somehow and tell the other library a transaction is occurring and then I am free to use the mobx API as intended.

**Comment by urugator** [comment]
If you're only interested about when the "transaction" ends, I think you could use [`reactionScheduler`](https://mobx.js.org/configuration.html#reactionscheduler-f---void--void)

**Reply by steveruizok** [reply]
Good tip! I'll try that.

**Comment by mweststrate** [comment]
Correct, in mobx-react we hook up react batches to mobx batches for similar
reasons:
https://github.com/mobxjs/mobx/blob/main/packages/mobx-react-lite/src/index.ts#L9
